frontend/src/pages/Author.jsx

```diff
@@ -56,8 +67,166 @@ class Author extends Component {
         });
     }
 
+    getFollowButton() {
+        let followbutton;
+        if (this.state.isSelf) {
+            return null;
+        }
+
+        else if (!this.state.isFollowing && !this.state.isSelf) {
+            followbutton = <Button positive onClick={this.sendFollowRequest}><Icon name = "user plus" />Follow</Button>
+        }
+
+        else if (this.state.isFollowing && !this.state.isSelf) {
+            followbutton = <Button negative onClick={this.sendUnfollowRequest}><Icon name = "user times"/>Unfollow</Button>
+        }
+
+        return <div>{followbutton}</div>;
+    
+    }
+
+    getloggedinAuthorIDandHost() {
+        const cookieauthorid = Cookies.get("userID"),
+            storeauthorid = store.getState().loginReducers.userId,
+            cookiehost = cookieauthorid.split("author/")[0],
+            storehost = store.getState().loginReducers.hostName;
+        let authorID;
+        let host;
+        if (cookieauthorid !== null) {
+            authorID = cookieauthorid;
+        } else if (storeauthorid !== null) {
+            authorID = storeauthorid;
+        }
+        if (cookiehost !== null) {
+            host = cookiehost;
+        } else if (storehost !== null) {
+            host = storehost;
+        }
+        return [authorID, host]
+    }
+
+    getFollowStatus() {
+        var authorID;
+        authorID = this.getloggedinAuthorIDandHost()
+        let urlPath = "/api/followers/" + utils.getShortAuthorId(this.props.location.state.fullAuthorId),
+            requireAuth = true;
+        HTTPFetchUtil.getRequest(urlPath, requireAuth)
+            .then((httpResponse) => {
+                if (httpResponse.status === 200) {
+                    httpResponse.json().then((results) => {
+                        for (let i = 0; i < results.authors.length; i++) {
+                            if (results.authors[i].id === authorID[0]){
+                                this.setState({
+                                    isFollowing: true
+                                })
+                            }
+                        }
+                    })
+                }
```

Comment:
Could add a toasty boi here for non-200 responses. 
But I approve anyways since everything else looks fine 👍